dashboard/app/controllers/admin_reports_controller.rb

```diff
@@ -217,50 +219,52 @@ def hoc_signups
     # Requested by Roxanne on 16 November 2015 to track HOC 2015 signups by day.
     authorize! :read, :reports
 
-    # Get the HOC 2014 and HOC 2015 signup counts by day, deduped by email and name.
-    # We restrict by dates to avoid long trails of (inappropriate?) signups.
-    data_2014 = DB[:forms].
-        where('kind = ? AND created_at > ? AND created_at < ?', 'HocSignup2014', '2014-08-01', '2015-01-01').
-        group(:name, :email).
-        # TODO(asher): Is this clumsy notation really necessary? Is Sequel
-        # really this stupid? Also below.
-        group_and_count(Sequel.as(Sequel.qualify(:forms, :created_at).cast(:date),:created_at_day)).
-        order(:created_at_day).
-        all.
-        map{|row| [row[:created_at_day].strftime("%m-%d"), row[:count].to_i]}
-    data_2015 = DB[:forms].
-        where('kind = ? AND created_at > ? AND created_at < ?', 'HocSignup2015', '2015-08-01', '2016-01-01').
-        group(:name, :email).
-        group_and_count(Sequel.as(Sequel.qualify(:forms, :created_at).cast(:date),:created_at_day)).
-        order(:created_at_day).
-        all.
-        map{|row| [row[:created_at_day].strftime("%m-%d"), row[:count].to_i]}
-
-    # Construct the hash {MM-DD => [count2014, count2015]}.
-    # Start by constructing the key space as the union of the MM-DD dates for
-    # data_2014 and data_2015.
-    require 'set'
-    dates = Set.new []
-    data_2014.each do |day|
-      dates.add(day[0])
-    end
-    data_2015.each do |day|
-      dates.add(day[0])
-    end
-    # Then populate the keys of our hash {date=>[count2014,count2015], ..., date=>[...]} with dates.
-    data_by_day = {}
-    dates.each do |date|
-      data_by_day[date] = [0, 0]
-    end
-    # Finally populate the values of our hash.
-    data_2014.each do |day|
-      data_by_day[day[0]][0] = day[1]
-    end
-    data_2015.each do |day|
-      data_by_day[day[0]][1] = day[1]
-    end
+    SeamlessDatabasePool.use_persistent_read_connection do
+      # Get the HOC 2014 and HOC 2015 signup counts by day, deduped by email and name.
+      # We restrict by dates to avoid long trails of (inappropriate?) signups.
+      data_2014 = DB[:forms].
```

Comment:
It's not anything right now... so you'd have to set it up.

The one place that particular constant is used:
https://github.com/code-dot-org/code-dot-org/blob/8d6b7ca3b39bfc6af1899ebe43e6b5ccf77a9eb0/bin/update-hoc-map
it's actually the readonly user for the master, but it would be better to
use the replica there.

(the changes in this PR lgtm so feel free to merge this and put all of
these on your todo list, btw...)

On Wed, Dec 2, 2015 at 1:45 PM, ashercodeorg <notifications@github.com>
wrote:

> In dashboard/app/controllers/admin_reports_controller.rb
> <https://github.com/code-dot-org/code-dot-org/pull/5795#discussion_r46480111>
> :
>
> > -    # Then populate the keys of our hash {date=>[count2014,count2015], ..., date=>[...]} with dates.
> > -    data_by_day = {}
> > -    dates.each do |date|
> > -      data_by_day[date] = [0, 0]
> > -    end
> > -    # Finally populate the values of our hash.
> > -    data_2014.each do |day|
> > -      data_by_day[day[0]][0] = day[1]
> > -    end
> > -    data_2015.each do |day|
> > -      data_by_day[day[0]][1] = day[1]
> > -    end
> > +    SeamlessDatabasePool.use_persistent_read_connection do
> > +      # Get the HOC 2014 and HOC 2015 signup counts by day, deduped by email and name.
> > +      # We restrict by dates to avoid long trails of (inappropriate?) signups.
> > +      data_2014 = DB[:forms].
>
> Is DB_READONLY a readonly version of the replica or a readonly version of
> the master DB?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/5795/files#r46480111>.
>